### ISSUE TITLE
Can you give a simple example of subgraph matching?

### ISSUE THREAD

**Issue by lihuiliullh** [sailab-code/gnn#5]
Hi GNN Authors:

May I know how you use GNN for subgraph matching? I notice that there two files named "GNN_Subgraph" and "Net_Subgraph". Are these files for subgraph matching? By the way, can you introduce some background knowledge about the dataset "sub_15_7_200"? I try to analyze this data file, but I found that it is different from what were described in the paper "The graph neural network model". 

Thanks a lot.

**Comment by mtiezzi** [comment]
Hi.
Yes, those two files are dedicated to the Subgraph Matching problem.
As described in the site [https://sailab.diism.unisi.it/gnn/tutorial.html#description](gnn), the user has to define its own Net file for the task he wants to attach.

Concerning the subgraph matching problem, the benchmark is designed to identify the nodes in the input graph that belong to a single subgraph given a priori during the learning phase. This is the same benchmark as in the original paper you indicated (the set is created using the original MATLAB code coming with the paper).

![image](https://user-images.githubusercontent.com/24276199/63098806-030ff600-bf74-11e9-935f-f866ac9f9d1b.png)



In this case, we created a dataset composed of 600 graphs, divided into 3 equal part (200 for training, 200 validation, 200 test).

Each graph is composed by 15 nodes, while the subgraph is composed of 7 nodes.

It is a node-focused task, in which you want to classify  each node as belonging or not to the target subgraph.

You can find a description also in our other [paper  ](https://link.springer.com/chapter/10.1007/978-3-319-99978-4_16)Rossi, A., Tiezzi, M., Dimitri, G.M., Bianchini, M., Maggini, M., & Scarselli, F. (2018).
"Inductive–Transductive Learning with Graph Neural Networks",
In Artificial Neural Networks in Pattern Recognition (pp.201-212).
Berlin : Springer-Verlag.

**Comment by lihuiliullh** [comment]
Hi GNN Authors:

Thanks.
I also notice that the code in Net_Clique.py, Net_Karate.py, Net_Mutag.py and Net_Subgraph.py are the same. Can I use Net_Subgraph.py for subgraph matching?
By the way, I run the code yesterday, and the accuracy is 0.5823333. It is a little lower than the result in the paper. May I know why?

Best

**Comment by mtiezzi** [comment]
The provided code are simple examples on toy datasets, with the simple purpose to show how to use our code.  
As I stated before, you can use the Net_Subgraph file for the subgraph problem.
The obtain better results, you can try to modify the hyperparameters.
In particular, you can play with the 
```python
learning_rate = 0.001  
state_dim = 10
```

in the GNN_Subgraph file (trying different sizes for the state or the learning rate) , and the 

```python
self.state_l1 = 5
self.output_l1 = 5

tf.layers.dense(inp, self.state_l1, activation=tf.nn.sigmoid)
```

dimensions and activation functions (note the **sigmoid** instead of the tanh)  in the Net_Subgraph file (you can also add more layers, as you wish).
I just updated the files with some hyperparameters that allow to reach better results in the Subgraph problem. 

If you play with the parameters and reach better results, please let me know and I will update the repo!
